### ISSUE TITLE
Add --ignore-installed flag to generated build command

### ISSUE THREAD

**Issue by mrvladus** [johannesjh/req2flatpak#68]
It's needed for some packages that Sdk has but Platform runtime doesn't. For example `python-lxml` is present in `org.gnome.Sdk` but not in `org.gnome.Platform`.

**Comment by johannesjh** [comment]
Thank you for the suggestion!

Some notes to help to proceed quickly: I think what you want to achieve is already possible in three different ways. I hope that one of the following solutions fits your need.

* You can already omit packages that you don't want to be installed (e.g., python-lxml because it is already present in the target platform) by omitting them when you invoke req2flatpak. Personally, that's how I am personally using req2flatpak.

* You can of course post-process the output of req2flatpak to suit your needs, in an automated or manual way.

* You can already customize flags such as `--ignore-installed` for the pip install command if using the python API instead of the commandline API, see the documentation https://johannesjh.github.io/req2flatpak/main/api.html#flatpakgenerator

...a fourth way, and probably what you've been asking for, would be to add a commandline option. Such an option would allow to specify custom flags for the `pip install` command. Is this what you are proposing?

How to proceed? I'd like to thank your for your proposal! And I'd be interested to hear what the other contributors think about this @real-yfprojects  @cbm755 ? My thoughts are: When creating req2flatpak, we sought to keep the CLI very simple. Flexibility is given through use of its python API. For this reason, in order to keep the commandline simple, I am somewhat opposed to, but not strictly against additional CLI options.

**Comment by jorio** [comment]
Hi @johannesjh,

> You can already omit packages that you don't want to be installed (e.g., python-lxml because it is already present in the target platform)

My use case for this is that `pygments` is installed in org.kde.Sdk, where my package is built, but it's not included in the actual runtime that the app will run on.  So, it would be nice to be able to specify `--ignore-installed` from the CLI without post-processing the yaml file.

**Comment by real-yfprojects** [comment]
> My thoughts are: When creating req2flatpak, we sought to keep the CLI very simple. Flexibility is given through use of its python API. For this reason, in order to keep the commandline simple, I am somewhat opposed to, but not strictly against additional CLI options.

I agree, the CLI is meant for the most common and basic use cases while the python API serves those who wish to further customize.

**Comment by johannesjh** [comment]
Thank you @mrvladus and @jorio  for explaining the rationale and usecase, and thank you @real-yfprojects for giving your thoughts. I suggest the following next steps:

I will detail the suggested solution, i.e., to use the python API for customization. I hope this provides useful help. To do so, I'll annotate the example from [req2flatpak's python api documentation](https://johannesjh.github.io/req2flatpak/main/api.html#example) for our specific usecase here this ticket:

```python3
# customization option 1, you can preprocess your requirements file. 
# e.g. using pip-compile, e.g., as in my side project favagtk, see for example:
# https://github.com/johannesjh/favagtk/tree/main/requirements

# then you read in the requirements file:
requirements = RequirementsParser.parse_file("requirements.txt")

# customization option 2, you can modify the list of requirements programatically, e.g.:
requirements = { r for r in requirements if really_needed(r) }

platforms = [PlatformFactory.from_string("cp310-x86_64")]
releases = PypiClient.get_releases(requirements)
downloads = {
    DownloadChooser.wheel_or_sdist(release, platform)
    for release in releases
    for platform in platforms
}

# customization option 3, you can provide a custom pip install template:
pip_install_template: str = "pip3 install --verbose --exists-action=i "
        '--no-index --find-links="file://${PWD}" '
        "--prefix=${FLATPAK_DEST} --no-build-isolation --ignore-installed "
build_module = FlatpakGenerator.build_module(requirements, downloads, pip_install_template=pip_install_template)

# customization option 4, you can tweak the output once it's been generated, e.g.:
build_module["build-commands"][0] = build_module["build-commands"][0].replace("--no-index", "--no-index --ignore-installed")
```

I will close this ticket for now. The reasons are that (so far) no-one stepped up to implement it, with two of us maintainers being rather sceptical (but not strictly opposed) to this change. To clarify: Should someone want to implement an option for the CLI which allows to pass additional commandline options to pip-install, I would review and merge this change.
